PowerEditor/src/ScitillaComponent/Buffer.cpp

```diff
@@ -879,7 +879,21 @@ bool FileManager::backupCurrentBuffer()
 				::SetFileAttributes(fullpath, dwFileAttribs);
 			}
 
-			FILE *fp = UnicodeConvertor.fopen(fullpath, TEXT("wbc"));
+			FILE *fp = UnicodeConvertor.fopen(fullpath, TEXT("rb+c")); // Use "r+" to NOT destroy existing file contents on open
+			if (fp)
+			{
+				// On "r+" open success move file position to the very beginning of file
+				if (_fseeki64(fp, 0, SEEK_SET))

```

Comment:
Here if `_fseeki64` failed to put file pointer on the beginning of file, you consider there's no more room on the hard disk, right?

Reply:
> Here if _fseeki64 failed to put file pointer on the beginning of file, you consider there's no more room on the hard disk, right?

No. I simply make sure `_fseeki64` does its job properly. If for some reason it doesn't, then we'll simply report write failure.

> According the description of MS fseek's document:
`On devices incapable of seeking, the return value is undefined.`

`On devices incapable of seeking, the return value is undefined.` are other types of streams (for example `stdin`). By `devices` its meant streams that are accessed strictly sequentially - no seeking possible for those.